Breaking change to remove GenerateConfig in minor release

Per the conversation in https://gophers.slack.com/archives/C3MSAFD40/p1538936074000100?thread_ts=1538924106.000100&cid=C3MSAFD40

Opening an issue here because, according to @stanislas-m, a breaking change was made in a minor release which resulted in old builds of buffalo breaking.

Here is a full repro of a build that previously worked that no longer works:

```
$ buffaloCommit=354657dfd81584bb82b8b6dff9bb9f6ab22712a8
$ depCommit=3e697f6afb332b6e12b8b399365e724e2e8dea7e
$ echo $buffaloCommit
354657dfd81584bb82b8b6dff9bb9f6ab22712a8
$ echo $depCommit
3e697f6afb332b6e12b8b399365e724e2e8dea7e
$ mkdir /tmp/gopath
$ export GOPATH=/tmp/gopath
$ export PATH=$GOPATH/bin:$PATH
$ cd /tmp/gopath
$ go get -u github.com/golang/dep/cmd/dep
$ cd src/github.com/golang/dep/cmd/dep
$ git checkout $depCommit
HEAD is now at 3e697f6a... Merge pull request #1832 from rousan/issue-1749-multiple-dir-in-gopath
$ go install
$ cd /tmp/gopath
$ go get -tags sqlite github.com/gobuffalo/buffalo
$ cd src/github.com/gobuffalo/buffalo
$ git checkout $buffaloCommit
HEAD is now at 354657d... Updated SHOULDERS.md
$ go get .
$ go install -tags sqlite
$ dep ensure
$ go test -tags sqlite ./...
# github.com/gobuffalo/buffalo/generators/soda
generators/soda/soda.go:37:11: undefined: generate.GenerateConfig
ok      github.com/gobuffalo/buffalo    0.196s
ok      github.com/gobuffalo/buffalo/binding    0.039s
ok      github.com/gobuffalo/buffalo/generators 0.051s
ok      github.com/gobuffalo/buffalo/generators/action  0.070s [no tests to run]
FAIL    github.com/gobuffalo/buffalo/generators/newapp [build failed]
ok      github.com/gobuffalo/buffalo/generators/resource        0.045s
ok      github.com/gobuffalo/buffalo/mail       0.104s
ok      github.com/gobuffalo/buffalo/middleware 0.170s
ok      github.com/gobuffalo/buffalo/middleware/basicauth       0.055s
ok      github.com/gobuffalo/buffalo/middleware/csrf    0.080s
ok      github.com/gobuffalo/buffalo/middleware/i18n    0.085s
ok      github.com/gobuffalo/buffalo/middleware/tokenauth       0.113s
ok      github.com/gobuffalo/buffalo/render     0.053s
ok      github.com/gobuffalo/buffalo/worker     0.014s
```

--- a/soda/cmd/generate/config_cmd.go
+++ b/soda/cmd/generate/config_cmd.go
@@ -48,6 +48,14 @@ var ConfigCmd = &cobra.Command{
 	},
 }
 
+// GenerateConfig generates pop configuration files.
+//
+// Deprecated: use github.com/gobuffalo/pop/genny/config instead.
+func GenerateConfig(cfgFile string, data map[string]interface{}) error {
+	oncer.Deprecate(0, "generate.GenerateConfig", "Use github.com/gobuffalo/pop/genny/config instead.")
+	return Config(cfgFile, data)
+}
+
 // Config generates pop configuration files.
 // Deprecated: use github.com/gobuffalo/pop/genny/config instead.
 func Config(cfgFile string, data map[string]interface{}) error {

